tests/unit/test_stubs.py

```diff
@@ -21,3 +23,52 @@ def testing_connect(*args):
             vcr_connection.cassette = Cassette("test", record_mode=mode.ALL)
             vcr_connection.real_connection.connect()
             assert vcr_connection.real_connection.sock is not None
+
+    def test_body_consumed_once_stream(self, tmpdir, httpbin):
+        self._test_body_consumed_once(
+            tmpdir,
+            httpbin,
+            "body_consumed_once_stream.yml",
+            BytesIO(b"1234567890"),
+            BytesIO(b"9876543210"),
+            BytesIO(b"9876543210"),
+        )
+
+    def test_body_consumed_once_iterator(self, tmpdir, httpbin):
+        self._test_body_consumed_once(
+            tmpdir,
+            httpbin,
+            "body_consumed_once_iterator.yml",
+            iter([b"1234567890"]),
+            iter([b"9876543210"]),
+            iter([b"9876543210"]),
+        )
+
+    def _test_body_consumed_once(
+        self,
+        tmpdir,
+        httpbin,
+        testfile,
+        data1,
+        data2,
+        data3,
+    ):
+        testpath = str(tmpdir.join(testfile))
+        host, port = httpbin.host, httpbin.port
+        match_on = ["method", "uri", "body"]
+        with use_cassette(testpath, match_on=match_on):
+            conn1 = httplib.HTTPConnection(host, port)
+            conn1.request("POST", "/anything", body=data1)
+            resp1 = conn1.getresponse()
+            assert resp1.status == 501  # Chunked request Not implemented
```

Comment:
Please help me see how the checking for `.status` helps the test. Can these three asserts be just dropped?

Reply:
Done.